### DISCUSSION TITLE
Interesting result when applying SGD_DGP demo

### DISCUSSION THREAD

**Discussion by findoctorlin** [cornellius-gp/gpytorch#2346] [General]
Hi everyone,

I was checking DGP example code from : [https://docs.gpytorch.ai/en/stable/examples/05_Deep_Gaussian_Processes/Deep_Gaussian_Processes.html](url)

In original code the epoch is 20 and learning rate is 0.01 for Adam algo, and the result visulization looks pretty good:
![image](https://github.com/cornellius-gp/gpytorch/assets/57867183/5aa9410e-bffe-4110-9016-6bbe3a137ce7)

training set: **Loss(Negative log likelihood)**: -0.6341
test set: **RMSE**: 0.09876643121242523, **NLL**: -0.6846765875816345

But when I use SGD with same epoch and learning rate(it doesn't make sense to make it same but the RMSE and NLL are similar as the training with Adam), the result visualization looks like:
<img width="561" alt="image" src="https://github.com/cornellius-gp/gpytorch/assets/57867183/15ae0166-963b-4521-83bd-db8168581103">
training set: **Loss(Negative log likelihood)**: 0.1388
test set: **RMSE**: 0.2517728805541992, **NLL**: 0.06959884613752365

 It looks like with SGD the model statiscally can converge itself, but the prediction is a flat line which looks like the DGP model is an underfitting model, but with Adam and SGD the loss on training on training dataset, the RMSE on test dataset and the NLL on test dataset looks quite similar, I wonder why it happens with SGD?
 
 Welcome discussion!

**Comment by gpleiss** [comment]
The adaptive learning rates of Adam are very necessary for GPs. We have even found this to be true for simple exact GPs.
